src/EcommerceStatsBundle/Bridge/Chameleon/BackendModule/EcommerceStatsBackendModule.php

```diff
@@ -0,0 +1,289 @@
+            StatsTableServiceInterface::DATA_GROUP_TYPE_MONTH => $this->translator->trans('chameleon_system_ecommerce_stats.date_month'),
+            StatsTableServiceInterface::DATA_GROUP_TYPE_WEEK => $this->translator->trans('chameleon_system_ecommerce_stats.date_week'),
+            StatsTableServiceInterface::DATA_GROUP_TYPE_DAY => $this->translator->trans('chameleon_system_ecommerce_stats.date_day'),
+        ];
+    }
+
+    /**
+     * @return never-returns - Uses `exit()` to finish the current request
+     */
+    protected function downloadTopsellers(): void
+    {
+        // header
+        $data = [[
+            $this->translator->trans('chameleon_system_shop.cms_module_shop_statistic.field_article_number'),
+            $this->translator->trans('chameleon_system_shop.cms_module_shop_statistic.field_article_name'),
+            $this->translator->trans('chameleon_system_shop.cms_module_shop_statistic.field_order_count'),
+            $this->translator->trans('chameleon_system_shop.cms_module_shop_statistic.field_value'),
+            $this->translator->trans('chameleon_system_shop.cms_module_shop_statistic.field_category'),
+        ]];
+
+        $topSellerItems = $this->topSellerService->getTopsellers(
+            $this->startDate,
+            $this->endDate,
+            $this->selectedPortalId
+        );
+
+        foreach ($topSellerItems as $topSellerItem) {
+            $data[] = [
+                $topSellerItem->getArticlenumber(),
+                $topSellerItem->getName(),
+                $this->local->FormatNumber($topSellerItem->getTotalOrdered(), 0),
+                $this->local->FormatNumber($topSellerItem->getTotalOrderedValue(), 2),
+                $topSellerItem->getCategoryPath(),
+            ];
+        }
+
+        $filename = $this->getCsvFilename('topseller');
+        CsvResponse::fromRows($filename, $data)->sendAndExit();
+    }
+
+    protected function getCsvFilename(string $basename): string
+    {
+        return sprintf('%s-%s-%s.csv',
+            $basename,
+            $this->local->FormatDate($this->startDate, TCMSLocal::DATEFORMAT_SHOW_DATE),
+            $this->local->FormatDate($this->endDate, TCMSLocal::DATEFORMAT_SHOW_DATE)
+        );
+    }
+
+    /**
+     * @return never-returns - Uses `exit()` to finish the current request
+     */
+    protected function getAsCsv(): void
+    {
+        $tableData = $this->stats->evaluate($this->startDate, $this->endDate, $this->dateGroupType, $this->showChange, $this->selectedPortalId);
+
+        $data = $this->csvExportService->getCSVData($tableData);
+        $filename = $this->getCsvFilename('stats');
+
+        CsvResponse::fromRows($filename, $data)->sendAndExit();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function GetHtmlHeadIncludes()
+    {
+        $includes = parent::GetHtmlHeadIncludes();
+
+        $cssPath = TGlobal::GetStaticURL('/bundles/chameleonsystemecommercestats/ecommerce_stats/css/ecommerce-stats.css');
```

Comment:
Is "GetStaticURL" actually needed?

At least after the symlinking below these should exist directly as those directories?

Reply:
Every other module I looked at used the method, so I don't see why this one shouldn't